sdk/core/azure-core/azure/core/rest/_rest.py

```diff
@@ -63,7 +62,7 @@
 
 ################################## CLASSES ######################################
 
-class HttpRequest(object):
+class HttpRequest(HttpRequestBackcompatMixin):
```

Comment:
add test for setting request attributes (i.e. set_json_body), and make sure accessing the request's properties with the new code should have the same behavior